### PR TITLE
Respect "expectingInput" in "inputHint"

### LINKED ISSUE #2022: InputHint not respected by WebChat in speech scenarios
InputHint not respected by WebChat in speech scenarios

## Screenshots
image

image
## Version
CDN: latest/webchat.js

## Describe the bug
The InputHint on a return activity indicates whether the microphone should be opened to take additional input. ExpectingInput should result in the microphone being opened but it's seemingly ignored.

## To Reproduce
- Open webchat with Bot
- Ask a question using speech that will trigger a response with InputHint set.

## Expected behavior
- Microphone to be opened automatically when `ExpectingInput` set on InputHint.

[Bug]

### PR DESCRIPTION
**Pull request by compulim**
Fixes #2022.

## Changelog Entry

-  Fix [#2022](https://github.com/Microsoft/BotFramework-WebChat/issues/2022). Fixed `"expectingInput"` in `inputHint` is not respected, by [@compulim](https://github.com/compulim) and [@corinagum](https://github.com/corinagum) in PR [#2149](https://github.com/Microsoft/BotFramework-WebChat/pull/2149)

## Description

Added code to handle `"expectingInput"` in `inputHint`.

Also tested against permutations of `expectingInput`, `acceptingInput`, and `ignoringInput` in consecutive cases:

- `expectingInput` will record on the microphone regardless if the last message was recorded from the microphone or not
- `acceptingInput` will record on microphone only if the last message was recorded from the microphone
- `ignoringInput` will stop the microphone, even if it is turned on

## Specific Changes

- `Dictation.js` is modified in a way that it will never record on microphone if they are any incoming activities (from the bot) is pending in the queue to speak

---

-  ~[ ] Testing Added~
   -  Tests will be added in later PR

### CODE REVIEW

File: packages/component/src/BasicSendBox.js [hunk 1]
```diff
@@ -66,8 +65,15 @@ BasicSendBox.propTypes = {
   })
 };
 
-export default connectToWebChat(({ dictateState, styleSet, webSpeechPonyfill }) => ({
-  dictationStarted: dictateState === STARTING || dictateState === DICTATING,
+// TODO: [P3] We should consider exposing core/src/definitions and use it instead
```
Comment (corinagum) [inline packages/component/src/BasicSendBox.js]: file issue?

File: packages/component/src/Dictation.js [hunk 2]
```diff
@@ -28,48 +29,69 @@ class Dictation extends React.Component {
...
+      // sendTyping();
 
-    // This is for two purposes:
-    // 1. Set send box will also trigger send typing
-    // 2. If the user cancelled out, the interim result will be in the send box so the user can update it before send
-    setSendBox(interims.join(' '));
+      // // This is for two purposes:
+      // // 1. Set send box will also trigger send typing
+      // // 2. If the user cancelled out, the interim result will be in the send box so the user can update it before send
+      // setSendBox(interims.join(' '));
```
Comment (corinagum) [inline packages/component/src/Dictation.js]: is this supposed to be commented out?

File: packages/component/src/Dictation.js [hunk 3]
```diff
@@ -28,48 +29,69 @@ class Dictation extends React.Component {
...
+        language,
+        numSpeakingActivities,
+        webSpeechPonyfill: { SpeechGrammarList, SpeechRecognition } = {}
+      },
       handleDictate,
       handleDictating,
       handleError
     } = this;
 
+    console.warn('!!!', {
```
Comment (corinagum) [inline packages/component/src/Dictation.js]: delete?

File: packages/core/src/sagas/markAllAsSpokenOnStopSpeakActivitySaga.js [hunk 4]
```diff
@@ -13,6 +13,7 @@ function* markAllAsSpoken() {
   }
 }
 
+// TODO: [P4] We should turn this into a reducer instead
```
Comment (corinagum) [inline packages/core/src/sagas/markAllAsSpokenOnStopSpeakActivitySaga.js]: file an issue? Or combine with the one above?

File: packages/core/src/sagas/stopSpeakingActivityOnInputSaga.js [hunk 5]
```diff
@@ -2,14 +2,14 @@ import { put, takeEvery } from 'redux-saga/effects';
...
-import { START_DICTATE } from '../actions/startDictate';
+// import { START_DICTATE } from '../actions/startDictate';
 import stopSpeakingActivity from '../actions/stopSpeakingActivity';
 import whileConnected from './effects/whileConnected';
 
 function* stopSpeakingActivityOnInput() {
   yield takeEvery(
     ({ meta, payload, type }) =>
-      type === START_DICTATE ||
+      // type === START_DICTATE ||
```
Comment (corinagum) [inline packages/core/src/sagas/stopSpeakingActivityOnInputSaga.js]: delete?

File: packages/core/src/sagas/stopSpeakingActivityOnInputSaga.js [hunk 6]
```diff
@@ -2,14 +2,14 @@ import { put, takeEvery } from 'redux-saga/effects';
 
 import { POST_ACTIVITY_PENDING } from '../actions/postActivity';
 import { SET_SEND_BOX } from '../actions/setSendBox';
-import { START_DICTATE } from '../actions/startDictate';
+// import { START_DICTATE } from '../actions/startDictate';
```
Comment (corinagum) [inline packages/core/src/sagas/stopSpeakingActivityOnInputSaga.js]: delete?

File: packages/component/src/BasicSendBox.js (hunk 1 above)
Comment (compulim) [reply packages/component/src/BasicSendBox.js]: I will leave the `TODO` inside the files for now. And we do a pass to copy all the TODO to GitHub Issues.

File: packages/component/src/Dictation.js (hunk 3 above)
Comment (compulim) [reply packages/component/src/Dictation.js]: 🤦‍♂️ Cleaned.

File: packages/core/src/sagas/markAllAsSpokenOnStopSpeakActivitySaga.js (hunk 4 above)
Comment (compulim) [reply packages/core/src/sagas/markAllAsSpokenOnStopSpeakActivitySaga.js]: We need a bot for these.

File: packages/core/src/sagas/stopSpeakingActivityOnInputSaga.js (hunk 6 above)
Comment (compulim) [reply packages/core/src/sagas/stopSpeakingActivityOnInputSaga.js]: 🤦‍♂️ Cleaned

File: packages/core/src/sagas/stopSpeakingActivityOnInputSaga.js (hunk 5 above)
Comment (compulim) [reply packages/core/src/sagas/stopSpeakingActivityOnInputSaga.js]: 🤦‍♂️ Cleaned

File: packages/component/src/Dictation.js (hunk 2 above)
Comment (compulim) [reply packages/component/src/Dictation.js]: Let me finish the "speech test harness" PR and will add the implementation + test here.

Comment (compulim) [reply packages/component/src/Dictation.js]: Fixed.

**Pull request merged**